### DISCUSSION TITLE
Split of the frontend code

### DISCUSSION THREAD

**Discussion by ctron** [guacsec/trustify#47] [Ideas]
Right now, I don't see any benefit of having the frontend code in the same repository. We don't share anything between the frontend and the backend anyway.

On the other hand, all the issues, PRs, GitHub warnings (sorted that out for now) feel like a massive distraction when focusing on backend topics. True, we need to deploy it together in the end. But right now, I don't see any overlap. And I also don't see the pros outweighing the cons here. If any of you see them, please point them out to me.

**Comment by helio-frota** [comment]
> Right now, I don't see any benefit of having the frontend code in the same repository. We don't share anything between the frontend and the backend anyway.

This is what I'm thinking since the beginning and already shared here and there...

I even gave `+1` for data transfer objects:

![2024-03-11_09-39](https://github.com/trustification/trustify/assets/6443576/71899666-fe45-4e87-a369-0ec2dd98bded)

But I still didn't understand the need to keep it in the same repository, after the `+1`.

> True, we need to deploy it together in the end.

Then -- Apparently this is the reason why both are together...
(Reason: some mitigation of deployment issues. ( is that correct? ) )
:man_shrugging: 

I believe a different repository will also help with front-end development speed-up 
mainly this point where --> apparently we need a fast approval ( yes-person ) model

**Reply by helio-frota** [reply]
I'm +1 to split front-ends ( whatever language/platform ) to a different repository.

We can see in this screenshot that spog (yellow) is almost "screaming" to separate from backend  :smile: 

![foo](https://github.com/trustification/trustify/assets/6443576/078367e1-6197-445d-93eb-c51f0d5fca8b)

* But makes a bit more sense since there is no DTOs :+1: 
* But now we have DTOs (with mocks)

**Comment by bobmcwhirter** [comment]
I'll +1 this.

@carlosthe19916 ?

**Reply by carlosthe19916** [reply]
Sounds good, I have no real preference in this matter. I'd just like to have this PR merged https://github.com/trustification/trustify/pull/48 (as this represents a first fully working Alpha1 version) and then I can migrate the code to its own repository
